### ISSUE TITLE
[BUG] pod probes not checked

### ISSUE THREAD

**Issue by 030** [stackrox/kube-linter#178]
It seems that the omission of live- and readiness checks is not detected.

**Comment by viswajithiii** [comment]
Hi @030, thanks for filing this issue! Can you give me an example YAML that you think should fail the check, but that isn't?

**Comment by 030** [comment]
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

```
apiVersion: v1
kind: Pod
metadata:
  name: goproxy
  labels:
    app: goproxy
spec:
  containers:
  - name: goproxy
    image: k8s.gcr.io/goproxy:0.1
    ports:
    - containerPort: 8080
    readinessProbe:
      tcpSocket:
        port: 8080
      initialDelaySeconds: 5
      periodSeconds: 10
    livenessProbe:
      tcpSocket:
        port: 8080
      initialDelaySeconds: 15
      periodSeconds: 20
```

If the either the readi- or liveness is omitted then an exit 1 should be returned indicating that the checks are missing.

```
apiVersion: v1
kind: Pod
metadata:
  name: goproxy
  labels:
    app: goproxy
spec:
  containers:
  - name: goproxy
    image: k8s.gcr.io/goproxy:0.1
    ports:
    - containerPort: 8080
```

**Comment by viswajithiii** [comment]
Hmm, that does seem to work as expected. See the output below.

Does your confusion perhaps stem from the fact that these checks are not enabled by default? They need to be explicitly enabled using a [config file](https://docs.kubelinter.io/#/configuring-kubelinter) (recommended) or by using the `--include` flag on the command line as I have done below for convenience.

```
kube-linter lint - --do-not-auto-add-defaults --include no-liveness-probe --include no-readiness-probe <<EOF
apiVersion: v1
kind: Pod
metadata:
  name: goproxy
  labels:
    app: goproxy
spec:
  containers:
  - name: goproxy
    image: k8s.gcr.io/goproxy:0.1
    ports:
    - containerPort: 8080
EOF
```

KubeLinter 0.2.1-0-gc53952b8e3

<standard input>: (object: <no namespace>/goproxy /v1, Kind=Pod) container "goproxy" does not specify a liveness probe (check: no-liveness-probe, remediation: Specify a liveness probe in your container. See https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/ for more details.)

```
<standard input>: (object: <no namespace>/goproxy /v1, Kind=Pod) container "goproxy" does not specify a readiness probe (check: no-readiness-probe, remediation: Specify a readiness probe in your container. See https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/ for more details.)
```

**Comment by 030** [comment]
I did not know I had to add:

```
--include no-liveness-probe --include no-readiness-probe
```

Should this not be a default setting?

**Comment by viswajithiii** [comment]
Some of our checks are enabled by default, while others are not. You can see the full list [here](https://docs.kubelinter.io/#/generated/checks), along with, for each check, whether it's enabled by default.

If we enable a check by default, we're trading off enforcement of better practices by default (good!) with whether we'll end up overwhelming people with too much noise when they run the tool (bad). So it's a judgement call. I did struggle with this specific one; while liveness and readiness probes are important, there are many configurations where they can be hard to set up for every single pod. Thanks for your input that this is something you'd have liked to see by default; I'll think more about it.

In general, our expectation was that when people use `kube-linter` in, say, a CI system, they will end up using a configuration file and tweaking the defaults, since everyone will have different checks they care about. For now, I recommend you do that in your specific case. Hope that makes sense, and thanks for filing this issu!
